packages/core/src/event-store/store.test.ts

```diff
@@ -639,6 +639,36 @@ test("insertFactoryLogFilterInterval inserts and merges child contract intervals
   expect(intervals).toMatchObject([[0, 1000]]);
 });
 
+test("insertFactoryLogFilterInterval handles topics", async (context) => {
+  const { eventStore } = context;
+
+  const factoryCriteria = {
+    address: "0xfactory",
+    eventSelector:
+      "0x0000000000000000000000000000000000000000000factoryeventsignature",
+    childAddressLocation: "topic1",
+    topics: [
+      "0x0000000000000000000000000000000000000000000factoryeventsignature",
+    ],
+  } satisfies FactoryCriteria;
+
+  await eventStore.insertFactoryLogFilterInterval({
+    chainId: 1,
+    factory: factoryCriteria,
+    block: blockOne,
+    transactions: blockOneTransactions,
+    logs: blockOneLogs,
+    interval: { startBlock: 0n, endBlock: 500n },
+  });
+
+  const intervals = await eventStore.getFactoryLogFilterIntervals({
+    chainId: 1,
+    factory: factoryCriteria,
+  });
```

Comment:
This test is fine, but it could be worth testing the inclusion rules. Like, if you insert a factory criteria with `topics: [[1, 2]]` and then request the same factory criteria but with `topics: [1]` the block ranges initially inserted should returned (because the initial request includes all logs present in the second).